packages/commonwealth/server/migrations/20221004191047-create-communityroles-roleassignments-and-update-threads-tables.js

```diff
@@ -0,0 +1,185 @@
+          deleted_at: { type: Sequelize.DATE, allowNull: true },
+          is_user_default: {
+            type: Sequelize.BOOLEAN,
+            allowNull: false,
+            defaultValue: false,
+          },
+        },
+        { transaction: t }
+      );
+
+      // Migrate CommunityRoles for Current Chains
+      const query = `
+      SELECT c.id as cid
+      FROM "Chains" c;`;
+      const chains = await queryInterface.sequelize.query(query, {
+        transaction: t,
+      });
+      await Promise.all(
+        chains[0].map(async (c) => {
+          // create CommunityRoles for each chain
+          await queryInterface.bulkInsert(
+            'CommunityRoles',
+            [
+              {
+                chain_id: c.cid,
+                name: 'admin',
+                created_at: new Date(),
+                updated_at: new Date(),
+              },
+            ],
+            { transaction: t }
+          );
+          await queryInterface.bulkInsert(
+            'CommunityRoles',
+            [
+              {
+                chain_id: c.cid,
+                name: 'moderator',
+                created_at: new Date(),
+                updated_at: new Date(),
+              },
+            ],
+            { transaction: t }
+          );
+          await queryInterface.bulkInsert(
+            'CommunityRoles',
+            [
+              {
+                chain_id: c.cid,
+                name: 'member',
+                created_at: new Date(),
+                updated_at: new Date(),
+              },
+            ],
+            { transaction: t }
+          );
+        })
+      );
+
+      // I think we need to do it half and half
+      // Migrate RoleAssignments for Current Roles
+      const role_query = `
+      SELECT r.id as rid, cr.name as crname, r.address_id, r.permission, r.chain_id, r.is_user_default, cr.id as crid
+      FROM "Roles" r
+      LEFT JOIN "CommunityRoles" cr
+      ON r.chain_id = cr.chain_id AND CAST(r.permission AS TEXT) = CAST(cr.name AS TEXT)
+      WHERE r.chain_id IN (SELECT "id" FROM "Chains") AND r.address_id IN (SELECT "id" FROM "Addresses");`;
+      const rolesWithCommunityRoles = await queryInterface.sequelize.query(
+        role_query,
+        {
+          transaction: t,
+        }
+      );
+      await Promise.all(
+        rolesWithCommunityRoles[0].map(async (r) => {
+          // create CommunityRoles for each chain
+          await queryInterface.bulkInsert(
+            'RoleAssignments',
+            [
+              {
+                community_role_id: r.crid,
+                address_id: r.address_id,
+                created_at: new Date(),
+                updated_at: new Date(),
+                is_user_default: r.is_user_default,
+              },
+            ],
+            { transaction: t }
+          );
+        })
+      );
+    });
+    // Add indexes
+    return queryInterface.sequelize.transaction(async (t) => {
```

Comment:
Why is this a separate transaction? just have one transaction for all the operations that need to be done...